[new product] matomo

nperez60 opened the pull request:
add matomo

nperez60 committed:
[new product] matomo

Add matomo, WIP
I Need to find out some information about LTS, and EOL schedule

nperez60 committed:
add the old name of matomo "piwik"

nperez60 committed:
add support and eol dates

nperez60 committed:
fix support date for releasecycle1

nperez60 committed:
add lts & php-runtime

nperez60 commented on products/matomo.md:
```diff
@@ -0,0 +1,76 @@
+    support: 2010-08-28
+    eol: true
+    latest: "0.9.9"
+    latestReleaseDate: 2010-08-27
```
Matomo doesn't have a clear documentation on support/eol date

V2 support date, based on this [blog post](https://matomo.org/blog/2017/12/piwik-2-reaches-end-life-soon-december-2017-update-now/)
V3 support date, baded on waybackmachine [here](https://web.archive.org/web/20220815000000*/https://fr.matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/)
V4 support date, based on another [blog post](https://fr.matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/)

Also, since V2, they introduce LTS releases, since that, all major version seems to be LTS. And  It seems to be a N-1 support strategy, where N-1 releases only receive critical bug or security patches.

The LTS part is more clearly documented [here](https://github.com/matomo-org/matomo/issues/8546), but some links mentioned in the issue are now in 404

BiNZGi replied:
Please include these details as comments and where useful (focus on supported releases) in the description section.

nperez60 replied:
done

nperez60 committed:
update product description

nperez60 committed:
remove localized URLs

nperez60 committed:
apply style guide

BiNZGi commented on products/matomo.md:
```diff
@@ -0,0 +1,77 @@
+---
+
+> [Matomo](https://matomo.org/) is an open-source analytics platform, is developed to track and analyze various components of digital activities
+>  including websites, servers, CMS, and online services.
```
A reference to Piwik in the text would be helpful:
```suggestion
> [Matomo](https://matomo.org/) (formerly known as [Piwik](https://matomo.org/blog/2018/01/piwik-is-now-matomo/)) is an open-source analytics platform.
> It is developed to track and analyze various components of digital activities
> including websites, servers, CMS, and online services.
```

BiNZGi commented on products/matomo.md:
```diff
@@ -0,0 +1,77 @@
+    support: 2010-08-28
+    eol: true
+    latest: "0.9.9"
+    latestReleaseDate: 2010-08-27
```
We only list stable cycles:
```suggestion
```

BiNZGi reviewed (changes requested):
First review with some suggestions...

nperez60 committed:
Update products/matomo.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: BiNZGi <BiNZGi@users.noreply.github.com>

nperez60 committed:
missing "methods", add identifiers, add comments, remove useless regex

nperez60 committed:
Merge branch 'endoflife-date:master' into nil-eol-matomo

nperez60 committed:
syntax

marcwrobel committed:
Minor updates

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+-   purl: pkg:github/matomo-org/matomo
+
+# support(x) = releaseDate(x+1)
+# eol(x) = releaseDate(x+2), confirm on https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/
```
```suggestion
# eol(x) documented on https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/
```

Looks like this article is always updated with the exact date, so no need to guess IMO.

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+releasePolicyLink: https://matomo.org/faq/new-to-piwik/faq_18925/
+changelogTemplate: https://github.com/matomo-org/matomo/releases/tag/__LATEST__
+activeSupportColumn: true
+releaseDateColumn: true
```
```suggestion
releaseDateColumn: true
activeSupportColumn: true
eolColumn: Long Term Support
```

I have mixed feeling about marking releases as LTS. Looking at https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/, it seems to be that "Long Term Support" is just the name of the security support phase.

So I propose the rename of the eolColumn and the removal of `lts` fields on each release. WDYT ?

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+> digital activities including websites, servers, CMS, and online services.
+
+Matomo follows an [N-1 support strategy](https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/),
+meanings that the 2 most recent major versions receive security and bug fixes.
```
```suggestion
Matomo only actively supports the latest version with new features, bug fixes, and security updates.
When a new major version is released, the previous major version enters the [Long Term Support
(LTS)](https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/)
phase and only receives critical bug and security fixes for at least 12 months.
```

It's better to use the same wording as on the Matomo website. WDYT ?

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+    lts: true
+    eol: false
+    latest: "5.0.1"
+    latestReleaseDate: 2024-01-05
```
```suggestion
    latest: "5.0.3"
    latestReleaseDate: 2024-03-07
```

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+    lts: true
+    eol: 2024-12-19 # https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/
+    latest: "4.16.0"
+    latestReleaseDate: 2023-11-28
```
```suggestion
    latest: "4.16.1"
    latestReleaseDate: 2024-02-05
```

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,74 @@
+    lts: true
+    eol: 2017-12-18 # https://matomo.org/blog/2017/12/piwik-2-reaches-end-life-soon-december-2017-update-now/
+    latest: "2.17.1"
+    latestReleaseDate: 2016-11-14
```
```suggestion
    latest: "2.18.1" # released to help upgrading to Matomo 4.x
    latestReleaseDate: 2020-07-01
```

marcwrobel reviewed (changes requested):
Thanks a lot for this contribution @nperez60, I have a few suggestions but overall LGTM.

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

nperez60 committed:
Update products/matomo.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

marcwrobel committed:
Drop lts fields

The LTS phase is true for all releases and is in fact the security support phase.

marcwrobel commented on products/matomo.md:
```diff
@@ -0,0 +1,73 @@
+
+# support(x) = releaseDate(x+1)
+# eol(x) documented on https://matomo.org/blog/2016/01/announcing-long-term-support-in-matomo-the-analytics-platform-for-your-mission-critical-projects/
+releases:
```
```suggestion
# No release is marked LTS, as this phase is true for all release and is considered here as the security support phase.
releases:
```

marcwrobel committed:
Update products/matomo.md

marcwrobel reviewed (approved):
LGTM, merging. Thanks @nperez60 !
